test/Data/Source/Spec.hs

```diff
@@ -27,13 +26,15 @@ prop desc f
 testTree :: Tasty.TestTree
 testTree = Tasty.testGroup "Data.Source"
   [ Tasty.testGroup "sourceLineRanges"
-    [ QC.testProperty "produces 1 more range than there are newlines" $
-      \ source -> QC.label (summarize source) $
-        length (sourceLineRanges source) QC.=== length (Text.splitOn "\r\n" (toText source) >>= Text.splitOn "\r" >>= Text.splitOn "\n")
-
-    , QC.testProperty "produces exhaustive ranges" $
-      \ source -> QC.label (summarize source) $
-        foldMap (`slice` source) (sourceLineRanges source) QC.=== source
+    [ testProperty "produces 1 more range than there are newlines" $ property $ do
+        source <- forAll (Gen.source (Hedgehog.Range.linear 0 100))
+        label (summarize source)
+        (length (sourceLineRanges source) === length (Text.splitOn "\r\n" (toText source) >>= Text.splitOn "\r" >>= Text.splitOn "\n"))
+
+    , testProperty "produces exhaustive ranges" $ property $ do
+        source <- forAll (Gen.source (Hedgehog.Range.linear 0 100))
+        label (summarize source)
```

Comment:
Ohhhh so _that’s_ what I was doing wrong. I was expecting it to be scoped, and misread the type signature.

Reply:
Yeah, I made the same mistake too! And to describe the error message that GHC gives (some nonsense about not being able to unify `m0` with `()` when the error is one of arity) as “disgustingly poor” is generous 😠